nixos/modules/programs/wayland/niri.nix

```diff
@@ -0,0 +1,81 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.niri;
+in
+{
+  options.programs.niri = {
+    enable = lib.mkEnableOption "Niri, a scrollable-tiling Wayland compositor";
+
+    package = lib.mkPackageOption pkgs "niri" { };
+
+    # Recommended by upstream
+    # https://github.com/YaLTeR/niri/wiki/Important-Software#portals
+    portalPackage = lib.mkPackageOption pkgs "xdg-desktop-portal-gnome" {
+      # https://github.com/NixOS/nixpkgs/pull/348193#discussion_r1798515394
+      extraDescription = "Note: `xdg-desktop-portal-gnome` is required for screencast support";
+    };
+
+    xwayland.enable = lib.mkEnableOption "XWayland support via xwayland-satellite";
+  };
+
+  config = lib.mkIf cfg.enable (
+    lib.mkMerge [
+      {
+        environment.systemPackages = [
+          cfg.package
+        ] ++ lib.optional cfg.xwayland.enable pkgs.xwayland-satellite;
+
+        services = {
+          displayManager.sessionPackages = [ cfg.package ];
+
+          # Recommended by upstream
+          # https://github.com/YaLTeR/niri/wiki/Important-Software#portals
+          gnome.gnome-keyring.enable = lib.mkDefault true;
+        };
+
+        systemd = {
+          packages = [ cfg.package ];
+
+          # Recommended by upstream
+          # https://github.com/YaLTeR/niri/wiki/Important-Software#authentication-agent
+          user.services.niri-polkit-agent = lib.mkIf config.security.polkit.enable {
```

Comment:
Could we add an additional enable option for this? I think there should be a way to disable this for people who want to manage this themselves.